packages/type-utils/tests/typeFlagUtils.test.ts

```diff
@@ -0,0 +1,94 @@
+import { parseForESLint } from '@typescript-eslint/parser';
+import type { TSESTree } from '@typescript-eslint/typescript-estree';
+import path from 'path';
+import * as ts from 'typescript';
+
+import { getTypeFlags, isTypeFlagSet } from '../src';
+import { expectToHaveParserServices } from './test-utils/expectToHaveParserServices';
+
+describe('typeFlagUtils', () => {
+  const rootDir = path.join(__dirname, 'fixtures');
+
+  function getType(code: string): ts.Type {
+    const { ast, services } = parseForESLint(code, {
+      project: './tsconfig.json',
+      filePath: path.join(rootDir, 'file.ts'),
+      tsconfigRootDir: rootDir,
+    });
+    expectToHaveParserServices(services);
+    const declaration = ast.body[0] as TSESTree.TSTypeAliasDeclaration;
+
+    return services.getTypeAtLocation(declaration.id);
+  }
+
+  describe('getTypeFlags', () => {
+    function runTestForAliasDeclaration(
+      code: string,
+      expected: ts.TypeFlags,
+    ): void {
+      const type = getType(code);
+      const result = getTypeFlags(type);
+      expect(result).toBe(expected);
+    }
+
+    it.each([
+      ['type Test = any;', 1],
+      ['type Test = unknown;', 2],
+      ['type Test = string;', 4],
+      ['type Test = number;', 8],
+      ['type Test = "text";', 128],
+      ['type Test = 123;', 256],
+      ['type Test = string | number', 12],
+      ['type Test = "text" | 123', 384],
+    ])('when code is "%s", type flags is %d', runTestForAliasDeclaration);
+  });
+
+  describe('isTypeFlagSet', () => {
+    function runTestForAliasDeclaration(
+      code: string,
+      flagsToCheck: ts.TypeFlags,
+      expected: boolean,
+    ): void {
+      const type = getType(code);
+      const result = isTypeFlagSet(type, flagsToCheck);
```

Comment:
Is the third argument 'isReceiver' in the function 'isTypeFlagSet' necessary?
Considering its usage in the callers of 'isTypeFlagSet', it doesn't seem to be utilized. Hence, I was wondering if it's safe to remove it.

Reply:
If it's never used in our code, you can mark it as deprecated and we can remove it later.

Reply:
Thanks! Added in 4d6d8ce .